app/use_cases/subscribe_to_protocol.rb

```diff
@@ -21,26 +21,32 @@ class SubscribeToProtocol < ActiveInteraction::Base
   # - protocol: the protocol to subscribe the person to (optional)
   # - person: the person to start the protocol for
   # - start_date: the date when the subscription should start
+  # rubocop:disable Metrics/AbcSize
   def execute
+    the_mentor = mentor
     the_protocol = find_protocol
     the_start_date = find_start_date
     Rails.logger.warn("Protocol #{the_protocol.id} does not have any measurements") if the_protocol.measurements.blank?
 
-    create_or_find_protocol_subscription(the_protocol, the_start_date)
+    # If we are a child, our parent is our mentor
```

Comment:
not a big fan of this tbh, but I see why you're doing it. Would it make sense to put this logic on the 'getter' for the mentor of a person? So we leave the field nil if there is no mentor, and we fill it with the mentor id only if we have a mentor? Not sure, but then we can still make a distinction (perhaps its completely unnecessary). What do you think?

Reply:
I think if you think back to the u-can-act student protocols. They had one protocol subscription, and it was filling_out_for their mentor. The reason being that in the questionnaire, we could then evaluate the {{mentor_name}} (oid) and so on, to make the questionnaires more personal.

I think the same applies here. The only reason we're setting the parents as mentor is that so we can maybe in the future use their name to make things more personal. I was thinking of e.g. just storing the first name of the parent, so in the subject line of the invitation mail to the kids we can write something like, "Your parent/caregiver, {{mentor_name}}, has invited you to join IKIA!," which could lead to higher response rates.

I think what you're saying makes more sense however, it would just require a little more effort on my part, which is probably why I didn't think of it :p I.e., just add a {{parent_name}} variable and have it evaluate to `person.parent.name`.

Reply:
Fixed.

I just removed setting the parent as mentor for now. We can add the {{parent}} variable later when we actually use it.